src/context.js

```diff
@@ -50,13 +56,48 @@ function EvolvContext(store) {
     }
   }
 
+  function mergePersistedContext(contextKey, context, storage) {
+    const persistedContext = storage.getItem(contextKey);
+    if (persistedContext) {
+      const parsedContext = JSON.parse(persistedContext);
+      if (context) {
+        return objects.deepMerge(context, parsedContext);
+      }
+
+      return parsedContext;
+    }
+
+    return context || {};
+  }
+
+  function mergeAllPersistedStorage(contextKey, context) {
+    if (window === undefined || !window.localStorage) {
+      return context || {};
+    }
+
+    const localStorageContext = mergePersistedContext(contextKey, context, window.localStorage);
+    return mergePersistedContext(contextKey, localStorageContext, window.sessionStorage);
+  }
+
+  function persistValue(key, value, local, sessionOnly) {
+    const storage = sessionOnly ? window.sessionStorage : window.localStorage;
+    const persistedContext = storage.getItem(local ? localStorageKey : remoteStorageKey);
+    const context = persistedContext ? JSON.parse(persistedContext) : {};
+    objects.setKeyToValue(key, value, context);
+    storage.setItem(local ? localStorageKey : remoteStorageKey, JSON.stringify(context));
+  }
+
   this.initialize = function(_uid, _remoteContext, _localContext) {
     if (initialized) {
       throw new Error('Evolv: The context is already initialized');
     }
     uid = _uid;
-    remoteContext = _remoteContext ? objects.deepClone(_remoteContext) : {};
-    localContext = _localContext ? objects.deepClone(_localContext) : {};
+
+    localStorageKey = 'evolv_' + this.uid + '_local_context';
+    remoteStorageKey = 'evolv_' + this.uid + '_remote_context';
```

Comment:
For consistency, our other storage entries use the prefix `evolv:`
```suggestion
    localStorageKey = 'evolv:' + this.uid + '_local_context';
    remoteStorageKey = 'evolv:' + this.uid + '_remote_context';
```

Reply:
And FWIW are camel-cased. So maybe this then:
```suggestion
    localStorageKey = 'evolv:' + this.uid + '_localContext';
    remoteStorageKey = 'evolv:' + this.uid + '_remoteContext';
```